auth/services/passport.js

```diff
@@ -0,0 +1,57 @@
+const passport = require('passport')
+const JwtStrategy = require('passport-jwt').Strategy
+const {ExtractJwt} = require('passport-jwt')
+const LocalStrategy = require('passport-local')
+const config = require('../../config')
+const User = require('../models/user')
+
+// Create local strategy
+const localOptions = {usernameField: 'email'}
+const localLogin = new LocalStrategy(localOptions, async (email, password, done) => {
+  // Verify this email and password, call done with the user
+  // if it is the correct email and password
+  // otherwise, call done with false
+  let user
+  try {
+    user = await User.findOne({email})
+  } catch (error) {
+    return done(error)
+  }
+  if (!user) {
+    return done(null, false)
+  }
+  // Compare passwords - is `password` equal to user.password?
+  user.comparePassword(password, (err, isMatch) => {
```

Comment:
This one cannot be promisified because this LocalStrategy definition is not invoked by any external code in a direct way. This project module passport.js is only required in order to tell passport package that this is the local strategy defined for this project

Reply:
but in this case comparePasswords is a schema method of User . Could be possible to have the method return a promise by itsslf, then await for the result? Afterwards, pass the result (or the caught error) to the done() callback

Reply:
That is already been done and about to be commited :D